### PR TITLE
Windows: Basic Chocolatey Packaging Processor

### PR DESCRIPTION
**Pull request by linuxfood**
This is based on top of #651, and should be merged after that one.

This PR:
* Adds a script that generates nuspec wrapper libraries to ease the generation of the chocolatey package .nuspec files. The generated output is committed to make usage reproducible.
* Wrapper classes that abstract the creation of `.nuspec` files, and `chocolateyInstall.ps1` files. Both are relatively naive right now, and will be improved as experience shows where features are missing.
* A processor that can build a simple chocolatey package from a recipe.  It supports both fetching from a URL at package install time, and copying an installer file into the generated package. Only the latter is unit tested.

### CODE REVIEW

File: Code/autopkglib/ChocolateyPackager.py [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+DefaultValue = VariableSentinal()
+
+
+class ChocolateyPackager(Processor):
+    """
+    Run `choco.exe` to build a single Nuget package.
+    """
+
+    description: str = __doc__
+    __name__: str = "ChocolateyPackager"  # workaround for bug in autopkglib.Processor
```
[the remark this answers is not in the document]
Comment (linuxfood) [reply Code/autopkglib/ChocolateyPackager.py:46]: Ah, you merged #652, which resolved issue #650. I'll remove this and in the next push.

File: Code/autopkglib/ChocolateyPackager.py [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+    CHOCO_CHECKSUM_TYPES,
+    CHOCO_FILE_TYPES,
+    ChocolateyInstallGenerator,
+    NuspecGenerator,
+)
+
+__all__ = ["ChocolateyPackager"]
+
+
+class VariableSentinal:
```
[the remark this answers is not in the document]
Comment (linuxfood) [reply Code/autopkglib/ChocolateyPackager.py:33]: Spelling is a myth. I'll fix this :)

**Comment by nmcspadden** [comment]
Is this PR rebased? I feel like a bunch of the changes to the precommit files, etc. were already merged in.

**Comment by linuxfood** [comment]
> Is this PR rebased? I feel like a bunch of the changes to the precommit files, etc. were already merged in.

Yes, sorry, there is one more tooling (hopefully final for awhile at least!) change in this PR. The purpose is to update `isort` so that comments can be added to source files to disable `isort` in specific regions. This makes it easier to create tools that are not sensitive to the working directory. Example: https://github.com/autopkg/autopkg/pull/653/commits/55956e4d17bde2b8f624f94f1cbc430d65ddcfec#diff-bf24321a12f0fa6d7b7ac72c27d51673R23-R29. If there is a better way to achieve this let me know, and the tooling update can be dropped from the changes.

After updating `isort`, I realized that I had updated `black` as well earlier, and so it seemed like avoiding the inevitable to not update `flake8`, so I did that too.

**Comment by linuxfood** [comment]
Updates for:
* bugs found while attempting to package Autopkg and Firefox for Windows.
* Autopkg package convenience script.

**Pull request merged**